source/modulo_new_core/include/modulo_new_core/communication/PublisherHandler.hpp

```diff
@@ -22,8 +27,31 @@ template<typename PubT, typename MsgT>
 PublisherHandler<PubT, MsgT>::PublisherHandler(PublisherType type, std::shared_ptr<PubT> publisher) :
     PublisherInterface(type), publisher_(std::move(publisher)) {}
 
+template<typename PubT, typename MsgT>
+void PublisherHandler<PubT, MsgT>::on_activate() {
+  if (this->get_type() != PublisherType::LIFECYCLE_PUBLISHER) {
```

Comment:
Is it okay if this throws or should it just be ignored?

Reply:
to be honest I think it should just warn and return

Reply:
We can't warn here though, except from using `std::cout`